src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1378,7 +1378,24 @@ void ImportCall(ILOpcode opcode, int token)
 
             if (opcode == ILOpcode.newobj)
             {
-                // TODO:
+                Check(method.IsConstructor, VerifierError.CtorExpected);
+                if (sig.IsStatic || methodType == null || method.IsAbstract)
+                {
+                    VerificationError(VerifierError.CtorSig);
+                }
+                else
+                {
+                    if (methodType.IsArray)
+                    {
+                        var arrayType = (ArrayType)methodType;
+                        Check(!IsByRefLike(StackValue.CreateFromType(arrayType.ElementType)), VerifierError.ArrayByRef);
```

Comment:
Should there be a similar rule in the else path for `newobj <byreflike-type>` ?

Reply:
I think `newobj <byref-like>` is verifiable. Or did you mean something else?

`newobj <byref-like>` is not common since many of those types do not have ctors.

The following will emit `newobj <byref-like>`   
```C#
using System;

class C
{
    static void Main(string[] args)
    {
        Test(new ArgIterator(new RuntimeArgumentHandle()));
    }

    static void Test(ArgIterator tr)
    {
    }
}
```

Reply:
You are right. I got confused.